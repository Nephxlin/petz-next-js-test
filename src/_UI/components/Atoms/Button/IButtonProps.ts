export interface IButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode
  stylepattern?: 'primary' | 'secondary'
  fullwidth?: boolean
}
