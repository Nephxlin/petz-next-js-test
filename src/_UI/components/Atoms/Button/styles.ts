import styled from "styled-components";
import { IButtonProps } from "./IButtonProps";

const stylePatternProps = {
  primary: {
    backgroundColor: '#E40F0F',
    color: '#FFFFFF',
    border: 'none',
    padding: '13px 30px',
    borderRadius: '50px',
    fontSize: '14px',
    fontWeight: 'bold',
    cursor: 'pointer',
    transition: 'all 0.5s cubic-bezier(0.250, 0.250, 0.750, 0.750)',
    '&:hover': {
      backgroundColor: '#FFFFFF',
      color: '#E40F0F',
      border: '0.5px solid #E40F0F',
    }
  },
  secondary:{
    backgroundColor: '#FFFFFF',
    color: '#1D1D1D',
    border: '1px solid #1D1D1D',
    fontSize: '12px',
    fontWeight: 'bold',
    padding: '13px 30px',
    cursor: 'pointer',
    borderRadius: '50px',
    '&:hover': {
      backgroundColor: '#1D1D1D',
      color: '#FFFFFF',
    }
  }
}


export const ButtonStyled = styled.button<IButtonProps>`
  ${({stylepattern}) => stylepattern === 'primary' && stylePatternProps.primary}
  ${({stylepattern}) => stylepattern === 'secondary' && stylePatternProps.secondary}
  ${({fullwidth}) => fullwidth ? `width:100%`: 'width: fit-content;'}

  &::disabled {
    opacity: 0.5;
    cursor: not-allowed;
  }
`