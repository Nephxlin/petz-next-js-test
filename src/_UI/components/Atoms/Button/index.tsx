import { IButtonProps } from './IButtonProps'
import * as S from './styles'


const Button = ({ children, stylepattern = 'primary', ...props }: IButtonProps) => {
  return (
    <S.ButtonStyled
      {...props}
      stylepattern={stylepattern}
      data-testid='button-styled-component'
    >
      {children}
    </S.ButtonStyled >
  )
}

export default Button