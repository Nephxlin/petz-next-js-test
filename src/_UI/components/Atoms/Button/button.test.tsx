
import { render, screen, } from '@testing-library/react';
import 'jest-styled-components';
import Button from './index';

// teste do componente Button
describe('Button component primary style', () => {

  it('renders Button component with primary style', () => {
    render(<Button stylepattern='primary'>test</Button>);
    const button = screen.getByTestId('button-styled-component');
    expect(button).toBeInTheDocument();
    expect(button).toHaveAttribute('stylepattern', 'primary');
    expect(button).toHaveStyleRule('background-color', '#E40F0F');
    expect(button).toMatchSnapshot();
  });

  it('renders Button component with secondary style', () => {
    render(<Button stylepattern='secondary' >test</Button>);
    const button = screen.getByTestId('button-styled-component');
    expect(button).toBeInTheDocument();
    expect(button).toHaveAttribute('stylepattern', 'secondary');
    expect(button).toHaveStyleRule('background-color', '#FFFFFF');
    expect(button).toMatchSnapshot();
  });
});