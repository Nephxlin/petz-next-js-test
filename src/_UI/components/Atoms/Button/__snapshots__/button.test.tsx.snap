// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component primary style renders Button component with primary style 1`] = `
.c0 {
  background-color: #E40F0F;
  color: #FFFFFF;
  border: none;
  padding: 13px 30px;
  border-radius: 50px;
  font-size: 14px;
  font-weight: bold;
  cursor: pointer;
  transition: all 0.5s cubic-bezier(0.250, 0.250, 0.750, 0.750);
  width: fit-content;
}

.c0:hover {
  background-color: #FFFFFF;
  color: #E40F0F;
  border: 0.5px solid #E40F0F;
}

.c0::disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

<button
  class="c0"
  data-testid="button-styled-component"
  stylepattern="primary"
>
  test
</button>
`;

exports[`Button component primary style renders Button component with secondary style 1`] = `
.c0 {
  background-color: #FFFFFF;
  color: #1D1D1D;
  border: 1px solid #1D1D1D;
  font-size: 12px;
  font-weight: bold;
  padding: 13px 30px;
  cursor: pointer;
  border-radius: 50px;
  width: fit-content;
}

.c0:hover {
  background-color: #1D1D1D;
  color: #FFFFFF;
}

.c0::disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

<button
  class="c0"
  data-testid="button-styled-component"
  stylepattern="secondary"
>
  test
</button>
`;
